--- src/Gopurs/CodeGen.purs
+++ src/Gopurs/CodeGen.purs
@@ -1025,6 +1025,8 @@
                       defMod = fromMaybe modNameStr (map unwrap mbMod)
                       key = defMod <> "." <> sanitizeName name
                       fromModuleArities = Map.lookup key moduleArities
+                      _ = unsafePerformEffect (Node.FS.Sync.appendTextFile Node.Encoding.UTF8 "/tmp/moduleArities.log" ("lookup " <> key <> " -> " <> show (map (\x -> "arity: " <> show x.arity <> ", fRet: " <> goTypeToStr x.fRet <> ", fArgs: " <> show (map goTypeToStr x.fArgs)) fromModuleArities) <> "\n"))
                     in case fromModuleArities of
                       Just { fullName, fArgs, fRet, arity } ->
                         let
