--- src/Gopurs/CodeGen.purs
+++ src/Gopurs/CodeGen.purs
@@ -762,9 +762,7 @@
                     if expectedGoType == TypeValue then res
                     else if isClosureNode a then res
                     else
-                      let
-                        _ = Debug.trace ("Typed coercion: " <> printGoExpr res.expr <> " from " <> goTypeToStr res.exprType <> " to " <> goTypeToStr expectedGoType) \_ -> unit
-                      in { stmts: res.stmts, expr: coerceGoExpr res.expr res.exprType expectedGoType, exprType: expectedGoType, nextId: res.nextId }
+                      { stmts: res.stmts, expr: coerceGoExpr res.expr res.exprType expectedGoType, exprType: expectedGoType, nextId: res.nextId }
           _, _ ->
             let res = translateExprImpl_ helpersRef depth modNameStr recVars moduleArities bound tcoIdent loopCtx isTail inEffectBlock nextId a
             in case res.exprType of
@@ -772,9 +770,7 @@
                 if expectedGoType == TypeValue then res
                 else if isClosureNode a then res
                 else
-                  let
-                    _ = Debug.trace ("Typed coercion (2): " <> printGoExpr res.expr <> " from " <> goTypeToStr res.exprType <> " to " <> goTypeToStr expectedGoType) \_ -> unit
-                  in { stmts: res.stmts, expr: coerceGoExpr res.expr res.exprType expectedGoType, exprType: expectedGoType, nextId: res.nextId }
+                  { stmts: res.stmts, expr: coerceGoExpr res.expr res.exprType expectedGoType, exprType: expectedGoType, nextId: res.nextId }
       Var (Qualified mbMn (Ident i)) ->
         let
           safeName = sanitizeName i
