--- src/Gopurs/CodeGen.purs
+++ src/Gopurs/CodeGen.purs
@@ -62,6 +62,8 @@
 boxGoExpr expr TypeBool = GoCall (GoSelector (GoVar "gopurs_runtime") "Bool") [ expr ]
 boxGoExpr expr (TypeStructPointer path) = GoConstructor "gopurs_runtime.Value" "" [ GoRaw "9", GoRaw (show (hashString ("Data_" <> String.replaceAll (Pattern ".") (Replacement "_") path))), GoCall (GoVar "unsafe.Pointer") [ expr ] ]
 boxGoExpr expr (TypeRecord _) = expr
+boxGoExpr expr (TypeInterface _) = expr
+boxGoExpr expr (TypeNativeArray _) = expr
 
 exprTypeToGoType :: ExprType -> GoType
 exprTypeToGoType Int = TypeInt64
@@ -92,6 +94,8 @@
     TypeString -> GoCall (GoSelector expr "StrVal") []
     TypeBool -> GoBinOp "!=" (GoSelector expr "IntVal") (GoInt 0)
     (TypeStructPointer path) -> GoTypeAssertion (GoSelector expr "UnsafePtr") ("*Data_" <> String.replaceAll (Pattern ".") (Replacement "_") path)
+    (TypeInterface _) -> expr
+    (TypeNativeArray _) -> expr
 
 getStructName :: String -> Maybe String -> String -> String
 getStructName modName mbMod ctorName = case mbMod of
