--- src/Gopurs/Printer.purs
+++ src/Gopurs/Printer.purs
@@ -79,6 +79,9 @@
 printGoExpr (GoMutate ident expr) =
   ident <> " = " <> printGoExpr expr
 
+printGoExpr (GoFuncLit params ret body) =
+  "func(" <> String.joinWith ", " (map (\(Tuple p goT) -> p <> " " <> goTypeToStr goT) params) <> ") " <> goTypeToStr ret <> " {\n" <> printGoExpr body <> "\n}"
+
 printGoExpr (GoIfElse cond trueBlock falseBlock) =
   "func() gopurs_runtime.Value {\n" <>
   "if " <> printGoExpr cond <> " {\n" <>
