--- src/Gopurs/CodeGen.purs
+++ src/Gopurs/CodeGen.purs
@@ -2160,7 +2160,11 @@
                             resFn = translateExprImpl_ helpersRef (depth + 1) modNameStr recVars moduleArities bound Nothing [] false false nextId flatFn
                             accArgs = foldl
                               ( \acc arg ->
                                   let
+                                    _unused = unsafePerformEffect (Console.log ("accArgs arg type: " <> printExprType (getExprType arg) <> " -> " <> goTypeToStr (exprTypeToGoType (unsafePerformEffect (Ref.read helpersRef)).pointerAdtPaths (unsafePerformEffect (Ref.read helpersRef)).enumAdts (unsafePerformEffect (Ref.read helpersRef)).elidedCtors modNameStr (getExprType arg))))
                                     argRes = translateExprImpl_ helpersRef (depth + 1) modNameStr recVars moduleArities bound Nothing [] false false acc.nextId arg
                                   in
                                     { stmts: acc.stmts <> argRes.stmts, exprs: Array.snoc acc.exprs argRes.expr, exprTypes: Array.snoc acc.exprTypes argRes.exprType, nextId: argRes.nextId }
                               )
