--- src/Gopurs/CodeGen.purs
+++ src/Gopurs/CodeGen.purs
@@ -793,8 +793,12 @@ translateExprImpl__ helpersRef depth modNameStr recVars moduleArities bound tcoI
         in case unwrapTcoExpr a, expectedGoType of
           Lit (LitRecord props), TypeStructPointer baseStructName fullName fullPath tArgs ->
             case Map.lookup fullName h.classDeclsFields of
               Just classInfo ->
                 let
+                  _ = Debug.Trace.trace ("Found classInfo for " <> fullName) \_ -> unit
                   classFields = classInfo.fields
@@ -803,7 +807,11 @@ translateExprImpl__ helpersRef depth modNameStr recVars moduleArities bound tcoI
                     in { stmts: accProps.stmts, expr: GoConstructor (hashString baseStructName) monoStructName typeArgsForDict accProps.exprs, exprType: expectedGoType, nextId: accProps.nextId }
               Nothing ->
+                let
+                  _ = Debug.Trace.trace ("NOT FOUND classInfo for " <> fullName <> " in LitRecord") \_ -> unit
+                in
                 let res = translateExprImpl_ helpersRef depth modNameStr recVars moduleArities bound tcoIdent loopCtx isTail inEffectBlock nextId tcoExpr
                 in { stmts: res.stmts, expr: boxGoExpr res.expr res.exprType, exprType: TypeValue, nextId: res.nextId }
