--- src/Gopurs/CodeGen.purs
+++ src/Gopurs/CodeGen.purs
@@ -1466,6 +1466,7 @@
               in { stmts: StmtEmpty, expr: funcExpr, exprType: TypeValue, nextId: resBody.nextId }
         else
           let
+            _ = unsafePerformEffect (if tcoIdent == Just "c100k" then Debug.trace ("UncurriedAbs else c100k resBody.exprType: " <> goTypeToStr resBody.exprType) \_ -> pure unit else pure unit)
             params = map fst paramsWithTypes
             makeCurried [] = resBody.expr
             makeCurried [p] = GoFunc p TypeValue TypeValue (GoBlock (flattenStmts resBody.stmts <> [ GoReturn (boxGoExpr resBody.expr resBody.exprType) ]))
