--- src/Gopurs/CodeGen.purs
+++ src/Gopurs/CodeGen.purs
@@ -949,6 +949,7 @@
                         Just mn -> String.replaceAll (Pattern ".") (Replacement "_") (unwrap mn)
                         Nothing -> modNameStr
                       fromModuleArities = if isLocal then Map.lookup name moduleArities else Nothing
+                      _ = unsafePerformEffect (if name == "fromInt" then Console.log ("fromInt fromModuleArities: " <> show (map (\x -> "arity: " <> show x.arity <> ", fRet: " <> goTypeToStr x.fRet) fromModuleArities)) else pure unit)
                       fromTypeSig = case extractFuncType flatFn of
                         Just { fArgs, fRet } ->
                           Just { fullName: "Call_" <> modPrefix <> "_" <> sanitizeName name, fArgs: map (exprTypeToGoType (unsafePerformEffect (Ref.read helpersRef)).pointerAdtPaths (unsafePerformEffect (Ref.read helpersRef)).enumAdts (unsafePerformEffect (Ref.read helpersRef)).elidedCtors modNameStr) fArgs, fRet: exprTypeToGoType (unsafePerformEffect (Ref.read helpersRef)).pointerAdtPaths (unsafePerformEffect (Ref.read helpersRef)).enumAdts (unsafePerformEffect (Ref.read helpersRef)).elidedCtors modNameStr fRet, arity: Array.length fArgs }
