--- src/Gopurs/CodeGen.purs
+++ src/Gopurs/CodeGen.purs
@@ -791,8 +791,9 @@ translateExprImpl__ helpersRef depth modNameStr recVars moduleArities bound tcoI
           expectedGoType = exprTypeToGoType (unsafePerformEffect (Ref.read helpersRef)).pointerAdtPaths (unsafePerformEffect (Ref.read helpersRef)).enumAdts (unsafePerformEffect (Ref.read helpersRef)).elidedCtors modNameStr type_
           h = unsafePerformEffect (Ref.read helpersRef)
 
         in
         let
           _ = case unwrapTcoExpr a of
-                Lit (LitRecord _) -> Debug.trace ("expectedGoType is " <> goTypeToStr expectedGoType <> " and type is " <> printExprType type_) \_ -> unit
+                Lit (LitRecord _) -> Debug.trace ("Got LitRecord: " <> show (unsafePerformEffect (Ref.read helpersRef)).pointerAdtPaths) \_ -> unit
                 _ -> unit
