--- src/Gopurs/CodeGen.purs
+++ src/Gopurs/CodeGen.purs
@@ -948,9 +948,9 @@
                         Just mn -> String.replaceAll (Pattern ".") (Replacement "_") (unwrap mn)
                         Nothing -> modNameStr
                       fromModuleArities = if isLocal then Map.lookup name moduleArities else Nothing
-                      _ = unsafePerformEffect (NodeFSSync.appendTextFile UTF8 "calls.log" ("Calling: " <> name <> " local: " <> show isLocal <> " fromModuleArities: " <> show (map (\x -> "arity: " <> show x.arity <> ", fRet: " <> goTypeToStr x.fRet) fromModuleArities) <> "\n"))
-                      fromTypeSig = unsafePerformEffect (NodeFSSync.appendTextFile UTF8 "calls.log" ("extractFuncType for " <> name <> " => " <> show (map (\x -> "fRet: " <> goTypeToStr x.fRet) (extractFuncType flatFn)) <> "\n")) `seq` case extractFuncType flatFn of
-                        Just { fArgs, fRet } ->
+                      _ = unsafePerformEffect (NodeFSSync.appendTextFile UTF8 "../altbak.pub/calls.log" ("Calling: " <> name <> " local: " <> show isLocal <> " fromModuleArities: " <> show (map (\x -> "arity: " <> show x.arity <> ", fRet: " <> goTypeToStr x.fRet) fromModuleArities) <> "\n"))
+                      fromTypeSig = case unsafePerformEffect (NodeFSSync.appendTextFile UTF8 "../altbak.pub/calls.log" ("extractFuncType for " <> name <> " => " <> show (map (\x -> "fRet: " <> goTypeToStr x.fRet) (extractFuncType flatFn)) <> "\n")) of
+                        _ -> case extractFuncType flatFn of
+                          Just { fArgs, fRet } ->
                           Just { fullName: "Call_" <> modPrefix <> "_" <> sanitizeName name, fArgs: map (exprTypeToGoType (unsafePerformEffect (Ref.read helpersRef)).pointerAdtPaths (unsafePerformEffect (Ref.read helpersRef)).enumAdts (unsafePerformEffect (Ref.read helpersRef)).elidedCtors modNameStr) fArgs, fRet: exprTypeToGoType (unsafePerformEffect (Ref.read helpersRef)).pointerAdtPaths (unsafePerformEffect (Ref.read helpersRef)).enumAdts (unsafePerformEffect (Ref.read helpersRef)).elidedCtors modNameStr fRet, arity: Array.length fArgs }
