--- src/Gopurs/CodeGen.purs
+++ src/Gopurs/CodeGen.purs
@@ -26,6 +26,7 @@
 import Effect.Unsafe (unsafePerformEffect)
 import Effect.Ref (Ref)
 import Effect.Ref as Ref
+import Node.FS.Sync as NodeFSSync
 import Gopurs.GoAst (GoFile, GoDecl, GoExpr(..), GoType(..), GoStmt(..), printGoDecl, printGoExpr, unboxGoExpr, boxGoExpr, goTypeToStr, stripNativeArray)
 import Node.Encoding (Encoding(..))
 import PureScript.Backend.Optimizer.CoreFn (Ann(..), Bind(..), Binder(..), Binding(..), CaseAlternative(..), CaseGuard(..), Comment, ConstructorType(..), DataConstructor, DataDecl, Expr(..), ExprType(..), Guard(..), Ident(..), Import(..), Literal(..), Meta(..), Module(..), ModuleName(..), Prop(..), ProperName(..), Qualified(..), ReExport, exprAnn, findProp, propKey, propValue, qualifiedModuleName, unQualified)
@@ -948,8 +949,8 @@
                         Just mn -> String.replaceAll (Pattern ".") (Replacement "_") (unwrap mn)
                         Nothing -> modNameStr
                       fromModuleArities = if isLocal then Map.lookup name moduleArities else Nothing
-                      _ = unsafePerformEffect (if name == "fromInt" then Node.FS.Sync.appendTextFile Node.Encoding.UTF8 "fromModuleArities.log" ("fromInt fromModuleArities: " <> show (map (\x -> "arity: " <> show x.arity <> ", fRet: " <> goTypeToStr x.fRet) fromModuleArities) <> "\n") else pure unit)
+                      _ = unsafePerformEffect (if name == "fromInt" then NodeFSSync.appendTextFile UTF8 "fromModuleArities.log" ("fromInt fromModuleArities: " <> show (map (\x -> "arity: " <> show x.arity <> ", fRet: " <> goTypeToStr x.fRet) fromModuleArities) <> "\n") else pure unit)
                       fromTypeSig = case extractFuncType flatFn of
