--- src/Gopurs/CodeGen.purs
+++ src/Gopurs/CodeGen.purs
@@ -792,7 +792,7 @@ translateExprImpl__ helpersRef depth modNameStr recVars moduleArities bound tcoI
         in
         let
           _ = case unwrapTcoExpr a of
-                Let _ _ _ _ -> unit
+                Abs args _ -> Debug.trace ("matched Abs for " <> name) \_ -> unit
                 _ -> unit
         in case unwrapTcoExpr a, expectedGoType of
           Lit (LitRecord props), TypeStructPointer baseStructName fullName fullPath tArgs ->
