--- src/Gopurs/CodeGen.purs
+++ src/Gopurs/CodeGen.purs
@@ -35,7 +35,7 @@
 import PureScript.Backend.Optimizer.Substitute (unify, substituteExprType, mapTcoExprTypes, substituteAst)
 import Effect.Ref as Ref
 import Node.FS.Sync as NodeFSSync
-import Gopurs.GoAst (GoFile, GoDecl, GoExpr(..), GoType(..), GoStmt(..), printGoDecl, printGoExpr, unboxGoExpr, boxGoExpr, goTypeToStr, stripNativeArray)
+import Gopurs.GoAst (GoFile, GoDecl, GoExpr(..), GoType(..), goTypeToStr)
 
 
 import Gopurs.Printer (printGoFile, printGoExpr, printGoDeclVar)
